vernac/attributes.mli

```diff
@@ -0,0 +1,102 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2018       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Vernacexpr
+
+type 'a attribute
+(** The type of attributes. When parsing attributes if an ['a
+   attribute] is present then an ['a] value will be produced.
+
+    Since attributes can be empty ['a] is always some option type. *)
+
+val unsupported_attributes : ?loc:Loc.t -> vernac_flags -> unit
+(** Error if any attributes remain in the flags. *)
+
+module Runner : sig
+  (** Notations to combine and run attributes. *)
+
+  val (++) : 'a attribute -> 'b attribute -> ('a * 'b) attribute
+  (** Combine 2 attributes. If any keys are in common an error will be raised. *)
+
+  val run : 'a attribute -> vernac_flags -> 'a * vernac_flags
+  (** Returns the unsupported attributes as well as the parsed value. *)
+
+  val (!) : 'a attribute -> vernac_flags -> 'a
+  (** Errors on unsupported attributes. *)
```

Comment:
This may be over-use of notations. Maybe this should be called `run` or `parse` and the `run` above should have some other name.

Reply:
I think so too.

Reply:
Changed.